### PR TITLE
ISPN-9469 Implement timeout for queries

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-9469

### CODE REVIEW

**Comment by gustavonalle** [comment]
CI is GREEN

File: query-core/src/main/java/org/infinispan/query/core/impl/BaseEmbeddedQuery.java [hunk 1]
```diff
@@ -82,19 +86,15 @@ public void resetQuery() {
          } else {
             Comparator<Comparable<FilterResult>[]> comparator = getComparator();
             if (comparator == null) {
-               results = StreamSupport.stream(spliterator(), false).collect(Collectors.toList());
+               results = StreamSupport.stream(spliterator(), false).collect(new TimedCollector<>(Collectors.toList(), timeout));
```
Comment (anistor) [inline query-core/src/main/java/org/infinispan/query/core/impl/BaseEmbeddedQuery.java:89]: So the timeout is detected when collecting. But what if the splitterator blocks indefinitely when trying to fetch next?

Comment (gustavonalle) [reply query-core/src/main/java/org/infinispan/query/core/impl/BaseEmbeddedQuery.java:89]: It will probably timeout after the RpcTimeout

Comment (gustavonalle) [reply query-core/src/main/java/org/infinispan/query/core/impl/BaseEmbeddedQuery.java:89]: I could also set the timeout in the stream itself, but only distributed streams support timeout

Comment (gustavonalle) [reply query-core/src/main/java/org/infinispan/query/core/impl/BaseEmbeddedQuery.java:89]: changed

File: query-dsl/src/main/java/org/infinispan/query/dsl/SearchTimeoutException.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.infinispan.query.dsl;
+
+/**
+ * Thrown when a query timeout occurs.
+ *
+ * @since 11.0
+ */
+public class SearchTimeoutException extends RuntimeException {
```
Comment (anistor) [inline query-dsl/src/main/java/org/infinispan/query/dsl/SearchTimeoutException.java:8]: I guess the days of dsl package are numbered. could we maybe move this to top level query package ?

Comment (gustavonalle) [reply query-dsl/src/main/java/org/infinispan/query/dsl/SearchTimeoutException.java]: moved

Comment (anistor) [reply query-dsl/src/main/java/org/infinispan/query/dsl/SearchTimeoutException.java]: Ok. Now I'm going to worry about split packages :)

**Comment by anistor** [comment]
Merged in master. Thanks @gustavonalle !

**Pull request closed without merging**